apps/web/src/components/ProfileSnapshot/utils.ts

```diff
@@ -0,0 +1,19 @@
+// versionHelpers.ts
+import { ComponentType } from "react";
+
+/**
+ * Returns the closest supported component ≤ requested version
+ */
+export function getSupportedVersionComponent<TProps>(
+  components: Record<number, ComponentType<TProps>>,
+  version: number,
+): ComponentType<TProps> | null {
+  const availableVersions = Object.keys(components)
+    .map(Number)
+    .sort((a, b) => b - a); // descending
+
+  for (const v of availableVersions) {
+    if (v <= version) return components[v];
+  }
+  return null;
```

Comment:
Shouldn't it return `1` or is this because there are `profile_snapshot` entries without the `version` key at all?

Reply:
Hmm, good point. I agree, thisis confusing.

I was alaways pasing in 1 with `snapshot.version ?? 1` as the fallback but it was not super clear, as noted here, so I moved the `1` to the function args top holefully clear things. The `null` should make more sense basically being the case of the current version not existing in the map so we have no component to actually render. eg

```
const map = {
  3: Version3,
  4: Version3
}

const snapshot1 = {
  version: null
}

const snapshot2 = {
  version: 2
}

// Returns null becuase defualt `1` not found
const Comp1 = getSupportedVersionedComponent(snapshot1.version)

// Returns null becuase version 2 is not in the map and all fallbacks also do not exist
const Comp2 = getSupportedVersionedComponent(snapshot2.version)
```